### PR TITLE
Set data directory to output path

### LINKED ISSUE #4765: CLI commands puts SQLite files in the nuget folder containing dotnet-ef
A relative path in the SQLite connection string + dotnet-ef  = the root for the relative path is the folder containing dotnet-ef.dll. This is because Microsoft.Data.Sqlite relies on AppContext.BaseDirectory, [which is Package.Current.InstalledLocation.Path](https://github.com/dotnet/corefx/blob/master/src/System.AppContext/src/System/AppContext.cs#L28).

image

To reproduce:
```c#
    public class TestContext : DbContext
    {
        protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
        {
            optionsBuilder.UseSqlite("Filename=./test.db");
        }
    }
```
```
dotnet ef migrations add First
dotnet ef database update
```

### PR DESCRIPTION
**Pull request by natemcmaster**
Fix #4765 

cc @bricelam

### CODE REVIEW

File: src/dotnet-ef/OperationExecutor.cs [hunk 1]
```diff
@@ -58,6 +61,13 @@ public OperationExecutor([CanBeNull] string startupProject, [CanBeNull] string e
 
             Reporter.Verbose.WriteLine("Build succeeded.".Bold().Black());
 
+            // TODO parameterize "configuration"
```
Comment (natemcmaster) [inline src/dotnet-ef/OperationExecutor.cs]: Is this already being tracked in an issue?

Comment (bricelam) [inline src/dotnet-ef/OperationExecutor.cs]: Nope. I think I saw this assumption--that design-time always uses `Debug`--in a few other places across the stack.

File: src/dotnet-ef/OperationExecutor.cs [hunk 2]
```diff
@@ -58,6 +61,13 @@ public OperationExecutor([CanBeNull] string startupProject, [CanBeNull] string e
...
+            var runtimeOutputPath = startupProjectContext.GetOutputPaths(DefaultConfiguration)?.RuntimeOutputPath;
+            if (!string.IsNullOrEmpty(runtimeOutputPath))
```
Comment (bricelam) [inline src/dotnet-ef/OperationExecutor.cs]: When is this `null`/empty?

File: src/dotnet-ef/OperationExecutor.cs [hunk 3]
```diff
@@ -58,6 +61,13 @@ public OperationExecutor([CanBeNull] string startupProject, [CanBeNull] string e
...
+            {
+                Environment.SetEnvironmentVariable(DataDirEnvName, runtimeOutputPath);
```
Comment (bricelam) [inline src/dotnet-ef/OperationExecutor.cs]: Can you add a TODO to call `AppDomain.SetData("DataDirectory", ...)` on `net45`? It sounds like this tool will be going that direction...

**Comment by bricelam** [comment]
:shipit:

File: src/dotnet-ef/OperationExecutor.cs (hunk 2 above)
Comment (natemcmaster) [inline src/dotnet-ef/OperationExecutor.cs]: I found odd corner cases, mostly due to restore errors. I think this is just due to the fact that we currently build against DependencyModel 1585. They've made lots of updates in the last 300 builds of cli.

**Comment by natemcmaster** [comment]
Updated & merged

**Pull request merged**